src/tyr/route_serializer_valhalla.cc

```diff
@@ -274,6 +277,19 @@ void legs(const valhalla::Api& api, int route_index, rapidjson::writer_wrapper_t
         writer.end_array();
       }
 
+      // Set bearings
+      // absolute bearing (degrees from north, clockwise) before and after the maneuver.
+      bool depart_maneuver = (maneuver_index == 0);
+      bool arrive_maneuver = (maneuver_index == directions_leg.maneuver_size() - 1);
+      if (!depart_maneuver) {
+        uint32_t in_brg = etp.GetPrevEdge(maneuver_index)->end_heading();
+        writer("bearing_before", static_cast<uint64_t>(in_brg));
+      }
+      if (!arrive_maneuver) {
+        uint32_t out_brg = maneuver.begin_heading();
+        writer("bearing_after", static_cast<uint64_t>(out_brg));
+      }
```

Comment:
In the osrm serializer, bearings default to `0` when not relevant. I found it confusing and choose to leave it empty.